plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/model/CodeModernizerArtifact.kt

```diff
@@ -52,6 +53,7 @@ open class CodeModernizerArtifact(
          * If anything goes wrong during this process an exception is thrown.
          */
         fun create(zipPath: String): CodeModernizerArtifact {
+            tempDir = createTempDirectory("codeTransformArtifacts-", UUID.randomUUID().toString())
```

Comment:
leaking state through the companion object. please have backlog for better encapsulation